kotest-framework/kotest-framework-multiplatform-plugin-gradle/src/main/kotlin/io/kotest/framework/multiplatform/gradle/KotestPluginExtension.kt

```diff
@@ -0,0 +1,13 @@
+package io.kotest.framework.multiplatform.gradle
+
+import org.gradle.api.provider.Property
+
+
+abstract class KotestPluginExtension {
+   /**
+    * The version to use for the Kotest compiler plugins.
+    *
+    * Defaults to [KOTEST_COMPILER_PLUGIN_VERSION].
+    */
+   abstract val kotestCompilerPluginVersion: Property<String>
+}
```

Comment:
Continuing @sksamuel's [comment...](https://github.com/kotest/kotest/pull/3156#issuecomment-1216493957)

> I also wouldn't call the field `kotestCompilerVersion` since that's not clear to me what it's setting. Is it the compiler plugin version?

Yeah, I'm happy to rename it. It's the version of either 
*  [`io.kotest:kotest-framework-multiplatform-plugin-embeddable-compiler`](https://search.maven.org/artifact/io.kotest/kotest-framework-multiplatform-plugin-embeddable-compiler)
* [`io.kotest:kotest-framework-multiplatform-plugin-legacy-native`](https://search.maven.org/artifact/io.kotest/kotest-framework-multiplatform-plugin-legacy-native)

Perhaps it makes sense to make these both separate properties. The legacy one can be marked as deprecated?

```suggestion
abstract class KotestPluginExtension {
  /** <insert docs here>, defaults to [KOTEST_COMPILER_PLUGIN_VERSION]. */ 
  abstract val kotestMultiplatformPluginEmbeddableCompilerVersion Property<String>
  /** <insert docs here>, defaults to [KOTEST_COMPILER_PLUGIN_VERSION]. */ 
  abstract val kotestMultiplatformPluginLegacyNativeVersion Property<String>
}
```

Reply:
I renamed it to `kotestEmbeddableCompilerVersion` to match the artifact name, `kotest-framework-multiplatform-plugin-embeddable-compiler`. 

I didn't add an additional property for the legacy compiler. It would be nice, but since it's slated for removal it's probably best to keep things slimmer. I'm pasting it here as a demo/in case it is useful.


```kotlin
   /**
    * The version of the **legacy** Kotest compiler plugin. This is only relevant for projects
    * that use Kotlin 1.6 or earlier.
    *
    * Defaults to [KOTEST_EMBEDDABLE_COMPILER_VERSION].
    */
   @Deprecated("Scheduled for removal https://youtrack.jetbrains.com/issue/KT-51301, update to Kotlin 1.7+")
   abstract val kotestLegacyEmbeddableCompilerVersion: Property<String>
```

Reply:
SInce we're moving to one, I think kotestCompilerPluginVersion is the best name.

Reply:
Done 👍